type-specs/Observable-spec.ts

```diff
@@ -0,0 +1,94 @@
+import { of, OperatorFunction } from 'rxjs';
+import { mapTo } from 'rxjs/operators';
+
+function a<I extends string, O extends string>(input: I, output: O): OperatorFunction<I, O>;
+function a<I, O extends string>(output: O): OperatorFunction<I, O>;
+function a<I, O extends string>(inputOrOutput: I | O, output?: O): OperatorFunction<I, O> {
+  return mapTo<I, O>(output === undefined ? inputOrOutput as O : output);
+}
+
```

Comment:
we actually just hit an issue where the following broke in one of our apps here at Google:

```ts
const customOperator = () => <T>(a: Observable<T>) => a;

of('string').pipe(customOperator());
```

NOTE: `customOperator` is `() => <T>(source: Observable<T>) => Observable<TT>` NOT `<T>() => (source: Observable<T>) => Observable<T>`, which seems to work fine.

Can we add a test for that here too?

Reply:
I narrowed it down to our use of 'noStrictGenerics' flag, which we should be using anyways. Anyways, good idea to add a test for it, though I expect it to just pass in your setup.